lib/resourceread/machineconfig.go

```diff
@@ -17,13 +19,37 @@ func init() {
 	}
 }
 
-// ReadMachineConfigV1OrDie reads MachineConfig object from bytes. Panics on error.
+// ReadMachineConfigV1 reads raw MachineConfig object from bytes. Returns MachineConfig and error.
+func ReadMachineConfigV1(objBytes []byte) (*mcfgv1.MachineConfig, error) {
+	if objBytes == nil {
+		return nil, errors.New("invalid machine configuration")
+	}
+
+	errDecode := errors.New("decode to machine config failed")
+
+	m, err := runtime.Decode(mcfgCodecs.UniversalDecoder(mcfgv1.SchemeGroupVersion), objBytes)
+	if err != nil {
+		return nil, err
+	}
+	if m == nil {
+		return nil, errDecode
+	}
+
+	mc, ok := m.(*mcfgv1.MachineConfig)
+	if !ok {
+		return nil, errDecode
```

Comment:
`runtime.Decode(mcfgCodecs.UniversalDecoder(mcfgv1.SchemeGroupVersion), objBytes)` will parse raw bytes of a type in https://github.com/openshift/machine-config-operator/blob/f915a1bfb2d98a2cdde2ea264008f7dfb58bf262/pkg/apis/machineconfiguration.openshift.io/v1/register.go#L40-L49

So this `m.(*mcfgv1.MachineConfig)` is that the raw byte was actually `MachineConfig` object. So the error should be `fmt.Errorf("expected *mcfvgv1.MachineConfig bute found %T", m)`

Reply:
Ah, indeed :+1: I've made the error more descriptive and updated the other others too.